### PR TITLE
Tweaks based on cppcheck linter warnings

### PR DESCRIPTION
**Pull request by aremmell**
- variable `foo` can be declared as pointer to const
- local variable `foo` shadows outer function
- the scope of the variable `foo` can be reduced
- condition `foo` is always true/false
- function argument 2 names different
- variable `foo` is assigned a value that is never used
- variable `foo` is reassigned a value before the old one has been used

### CODE REVIEW

File: tests/tests.c [hunk 1]
```diff
@@ -1775,9 +1773,11 @@ bool sirtest_filesystem(void) {
         1234
     };
 
+#if defined(__WIN__)
     if (get_wineversion()) {
```
Comment (johnsonjh) [inline tests/tests.c:1777]: I don't think this is correct - we won't be running Wine on Windows.

**Review by johnsonjh — APPROVED** [review]

**Review by aremmell — COMMENTED** [review]
👍

**Review by johnsonjh — APPROVED** [review]

**Review by johnsonjh — DISMISSED** [review]
👍🏻

**Review by johnsonjh — APPROVED** [review]
😑

**Pull request merged**